cmd/podman/machine/init.go

```diff
@@ -29,6 +29,7 @@ var (
 	initOpts           = machine.InitOptions{}
 	defaultMachineName = machine.DefaultMachineName
 	now                bool
+	defaultProvider    = GetSystemDefaultProvider()
```

Comment:
Is there a reason to move this into init()? We should avoid calling functions in init because they have to be executed for everything, so things like `podman --help`,  `podman run` now call code that is never required for them. I know the existing code has a lot of these bad practises but we should avoid adding new ones. Also https://github.com/containers/podman/pull/17639 will change the function,

Reply:
fair point, I updated this to a better approach